test/e2e/main_test.go

```diff
@@ -212,112 +243,65 @@ func testInstall(
 	meteringOperatorImageTag,
 	testName string,
 	targetPods int,
+	testInstallFunctions []InstallTestCase,
 	spec metering.MeteringConfigSpec,
 	namespacePrefix string,
 	testOuputPath string,
 	cleanupScriptPath string,
 ) {
 	// create a directory used to store the @testName container and resource logs
-	testOutputDir := filepath.Join(testOuputPath, testName)
-	err := os.Mkdir(testOutputDir, 0777)
+	testCaseOutputBaseDir := filepath.Join(testOuputPath, testName)
+	err := os.Mkdir(testCaseOutputBaseDir, 0777)
 	assert.NoError(t, err, "creating the test case output directory should produce no error")
 
-	// randomize the namespace to avoid existing namespaces
 	rand.Seed(time.Now().UnixNano())
-	namespace := namespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
 
-	deployerCtx, err := df.NewDeployerCtx(namespace, meteringOperatorImageRepo, meteringOperatorImageTag, reportingOperatorImageRepo, reportingOperatorImageTag, spec, testOutputDir, targetPods)
-	assert.NoError(t, err, "creating a new deployer context should produce no error")
+	for _, testFunction := range testInstallFunctions {
+		// randomize the namespace to avoid existing namespaces
+		namespace := namespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
```

Comment:
I believe we include the testcase name here in some way as it is we can still get conflicts and it's only 50 unique combos. Maybe `namespace := fmt.Sprintf("%s-%s-%d", namespacePrefix, testCaseName, rand.Intn(50))` We may need to truncate/limit it to `63` characters as that's the limit on kube NS names.

Reply:
Good point - we can first try with the combination above, and if the length of that string exceeds the kube ns limit, retry with the original implementation.

I took a look at the kube namespace validation regex, and it only allows lowercase characters and numerics.